### PR TITLE
fix decal in aria

### PR DESCRIPTION
**Pull request by Aman1919**
## Brief Information

This pull request is in the type of:

- [x] bug fixing
- [ ] new feature
- [ ] others

### What does this PR do?
fix issue - https://github.com/apache/echarts/issues/18118#issue-1512698774
"series-sunburst.data.itemStyle.decal" is able to override "aria.decal. decals"

## Document Info

One of the following should be checked.

- [x] This PR doesn't relate to document changes
- [ ] The document should be updated later
- [ ] The document changes have been made in apache/echarts-doc#xxx

### ZRender Changes

- [ ] This PR depends on ZRender changes (ecomfe/zrender#xxx).

### Related test cases or examples to use the new APIs

N.A.

### Merging options

- [ ] Please squash the commits into a single one when merging.

### CODE REVIEW

File: src/data/Tree.ts [hunk 1]
```diff
@@ -265,6 +265,9 @@ export class TreeNode {
     getVisual(key: string): unknown {
         return this.hostTree.data.getItemVisual(this.dataIndex, key as any);
     }
+     getDecal(key: string): unknown {
```
Comment (Ovilia) [inline src/data/Tree.ts:268]: Why is `key` necessary` here? It's always `'decal'`, right?

**Review by Ovilia — CHANGES_REQUESTED** [review]
Thanks for your contribution. Please also add a test case for this issue. Please checkout [the wiki](https://github.com/apache/echarts/wiki/How-to-make-a-pull-request#test) for more information.

**Comment by Aman1919** [comment]
added a test case for the issue

**Review by Ovilia — CHANGES_REQUESTED** [review]
image

The test result doesn't seem to be right to me. It should use circle and rect as symbols, right?

**Comment by Aman1919** [comment]
i did some changes and after that result is looking like this
image

File: src/chart/sunburst/SunburstSeries.ts [hunk 2]
```diff
@@ -164,7 +164,7 @@ class SunburstSeriesModel extends SeriesModel<SunburstSeriesOption> {
     static readonly type = 'series.sunburst';
     readonly type = SunburstSeriesModel.type;
 
-    ignoreStyleOnData = true;
+    ignoreStyleOnData = false;
```
Comment (Ovilia) [inline src/chart/sunburst/SunburstSeries.ts:167]: Remove this line and `ignoreStyleOnData` is set to be `false` by default.

File: src/visual/aria.ts [hunk 3]
```diff
@@ -82,15 +82,10 @@ export default function ariaVisual(ecModel: GlobalModel, api: ExtensionAPI) {
                 if (ecModel.isSeriesFiltered(seriesModel)) {
                     return;
                 }
-                if (zrUtil.isFunction(seriesModel.enableAriaDecal)) {
```
Comment (Ovilia) [inline src/visual/aria.ts:85]: Do not remove these lines. You can remove the `enableAriaDecal` in `SunburstSeries.ts`.

File: src/visual/aria.ts [hunk 4]
```diff
@@ -82,15 +82,10 @@ export default function ariaVisual(ecModel: GlobalModel, api: ExtensionAPI) {
...
-                    // Let series define how to use decal palette on data
-                    seriesModel.enableAriaDecal();
-                    return;
-                }
 
                 const data = seriesModel.getData();
 
-                if (!seriesModel.isColorBySeries()) {
+                if (seriesModel.isColorBySeries()) {
```
Comment (Ovilia) [inline src/visual/aria.ts:88]: This is not correct because this will break decal for other series.

**Review by Ovilia — CHANGES_REQUESTED** [review]

**Comment by Aman1919** [comment]
@Ovilia did some changes and added another case

File: src/visual/aria.ts [hunk 5]
```diff
@@ -82,12 +82,10 @@ export default function ariaVisual(ecModel: GlobalModel, api: ExtensionAPI) {
                 if (ecModel.isSeriesFiltered(seriesModel)) {
                     return;
                 }
-                if (zrUtil.isFunction(seriesModel.enableAriaDecal)) {
-                    // Let series define how to use decal palette on data
-                    seriesModel.enableAriaDecal();
+            if (zrUtil.isFunction(seriesModel.enableAriaDecal)) {
```
Comment (Ovilia) [inline src/visual/aria.ts:85]: Please revert the changes (indent) here.

File: src/data/Tree.ts [hunk 6]
```diff
@@ -265,7 +265,6 @@ export class TreeNode {
     getVisual(key: string): unknown {
         return this.hostTree.data.getItemVisual(this.dataIndex, key as any);
     }
```
Comment (Ovilia) [inline src/data/Tree.ts:267]: Please revert the change here.

File: test/sunburst-ariaDecal2.html [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+            #main {
+                width: 100%;
+                height: 100%;
+                /* background-image: url(old_mathematics.png); */
+                /* background-repeat: repeat; */
+            }
+        </style>
+        <div id="main"></div>
+        <script>
+            var chart;
```
Comment (Ovilia) [inline test/sunburst-ariaDecal2.html:39]: The test result of this case is not correct. Since "Grandpa", "Nancy" and "Mike" have different colors, they should also have different decals, otherwise, decal is not useful at all.
Besides, this case can be put into `sunburst-ariaDecal.html` rather than in a seperate file.

**Review by Ovilia — CHANGES_REQUESTED** [review]

**Review by Ovilia — CHANGES_REQUESTED** [review]
How can Sunburst items have the same decal if they have the same color after this fixing?

**Comment by Aman1919** [comment]
After fixing result was looking like this 
image
mike, nancy and grandpa decal are working 
do you mean their children should also have their decal ?

**Comment by Ovilia** [comment]
@Aman1919 Yes, in the above image, if two items have the same color, they should have the same decal.

**Comment by Aman1919** [comment]
> @Aman1919 Yes, in the above image, if two items have the same color, they should have the same decal.

I think color is being added by value of children items . is this not correct Color shouldn't be added like  this ?

**Comment by Ovilia** [comment]
@Aman1919 By default, the color should act like this. Color relating to value is only used when visualMap is used.

**Comment by Aman1919** [comment]
I did changes after that result is looking like this 
image
but it giving default backgroundcolor blue . I checked in pie and treemap they were also giving default blue  .  is this will be correct ?

**Comment by Ovilia** [comment]
No. This will break the original design of sunburst.

**Comment by Aman1919** [comment]
i did changes and result is looking like this . is this will be correct ?
image

**Review by Ovilia — CHANGES_REQUESTED** [review]
No. As I said, the colors and decals should be relevant. The colors/decals of the first-level nodes should be different and nodes of other levels should have the same color/decal as their ancestor of the first level.

**Comment by Aman1919** [comment]
so, lower levels node should be of lighter color than the upper level ? Could you suggest me something ?

**Comment by Ovilia** [comment]
@Aman1919 Please check [this demo](https://echarts.apache.org/examples/zh/editor.html?c=sunburst-simple). The colors are correct.

**Comment by Aman1919** [comment]
@Ovilia 
image
